lib/internal/readline/interface.js

```diff
@@ -441,6 +459,16 @@ class Interface extends InterfaceConstructor {
     }
   }
 
+  // Convert newlines to a consistent format for history storage
+  [kNormalizeHistoryLineEndings](line) {
+    return StringPrototypeReplaceAll(line, '\n', '\r');
+  }
+
+  // Convert newlines back to the format expected by the UI
+  [kDenormalizeHistoryLineEndings](line) {
+    return StringPrototypeReplaceAll(line, '\r', '\n');
+  }
```

Comment:
What about consolidating the methods by accepting two more arguments that would indicate the changes arguments?

That way we only need a single method and we can use default values for the normalization and just denormalization uses the args? I understand if you rather want it explicit though, so it's just a suggestion.